### ISSUE TITLE
Using "exclude" instead of "filesGlob" in tsconfig.json produces large amounts of pending requests.

### ISSUE THREAD

**Issue by ralfbiedert** [TypeStrong/atom-typescript#634]
When I move from `filesGlob` to `excludes` in my `tsconfig.json` I get lots of pending requests that take long to resolve, even for tiny stub projects containing only 2 files. For larger projects Atom becomes unusable. 

<img width="905" alt="screen shot 2015-10-01 at 14 44 15" src="https://cloud.githubusercontent.com/assets/301288/10221007/ab5c3604-684b-11e5-9c3c-d96e3b6fe3d6.png">

<img width="341" alt="screen shot 2015-10-01 at 14 52 54" src="https://cloud.githubusercontent.com/assets/301288/10221064/1ffa220a-684c-11e5-852b-1d43598bacb8.png">

Once I revert to `filesGlob` and restart Atom things are back to normal. 

This bug could be related to https://github.com/TypeStrong/atom-typescript/issues/624.

I am using Atom 1.0.19, atom-typescript 7.0.2, Typescript 1.7.0-dev.20151001 on Mac OS 10.11, however I could observe this behavior for older Atom versions, atom-typescript 5.x and Typescript 1.5.

**Comment by basarat** [comment]
`exclude` wasn't supported till a day ago : https://github.com/TypeStrong/atom-typescript/releases/tag/v7.0.3 

So try again, and comment if it still doesn't work :rose:

**Comment by ralfbiedert** [comment]
Just checked with 7.1.3 and I still get the pending requests in my big project:

![screen shot 2015-10-05 at 10 37 24](https://cloud.githubusercontent.com/assets/301288/10276071/323df744-6b4d-11e5-97f8-8318514bb05a.png)

**Comment by nycdotnet** [comment]
Note - I wound up with a perf problem in grunt-ts when I implemented `exclude` as well.  Adding a "not glob" doesn't actually help because the glob matcher isn't clever enough to optimize away that work.  It goes searching through all the excluded folders and then ignores whatever it finds.

Here was my fix:
https://github.com/TypeStrong/grunt-ts/commit/cad78ccca5797e4e8d6db1a7a81305979a64ee6a

Basically I add *.ts and *.tsx in the current folder.  Then I enumerate the names of the subfolders in the folder containing `tsconfig.json`.  Then I remove the excluded subfolders and then add **/*.ts and **/*.tsx for the remaining ones.

**Comment by blakeembrey** [comment]
@ralfbiedert Can you confirm whether this is still an issue on the latest release? It's since moved to a more effective glob.

**Comment by ralfbiedert** [comment]
Did a very quick check with 7.4.1 and it seems to be fixed. The number of pending requests briefly spikes up to 20, but returns to 0 after about 1s.

**Comment by ralfbiedert** [comment]
Hm, I don't know if this is a regression since 7.4.1, but I just tried 7.4.4 with another big project and again I got a number of pending requests that don't go away. 

<img width="843" alt="screen shot 2015-10-21 at 10 49 16" src="https://cloud.githubusercontent.com/assets/301288/10631968/9d8ab69a-77e1-11e5-82fc-d1d39fa9f08a.png">

**Comment by blakeembrey** [comment]
@ralfbiedert Is the `exports` key a typo?

**Comment by blakeembrey** [comment]
Actually, I have the same issue on my project too - I just tried going back multiple versions when it was previously working and it seems I have the same issue still, not sure where or how this got introduced now.

**Comment by blakeembrey** [comment]
@basarat Do you have an idea of how to debug the slowness? I can't seem to narrow it down by just throwing out guesses and globbing should not be this slow (ignore is definitely still working outside of Atom). Is there a way to `console.log` inside whatever process is running this?

**Comment by basarat** [comment]
> Is there a way to console.log inside whatever process is running this?

Yup `console.error`. See : https://github.com/TypeStrong/atom-typescript/blob/master/CONTRIBUTING.md#debugging

Note: There are only two processes ... one managed by atom aka. the atom window that just calls our functions ... and one we run in the background so as to not block the window thread.

**Comment by blakeembrey** [comment]
Awesome, thanks :+1: By "whatever process" I was referring to however it was managed, not the number of them, but yeah - where does the different process get started from? :smile:

**Comment by basarat** [comment]
> where does the different process get started from?

Here [in parent.ts](https://github.com/TypeStrong/atom-typescript/blob/9740e2bb3ee1ae60f24167800886c2769ded1cd3/lib/worker/parent.ts#L28)

Its a bit of a mess ... but there wasn't a good OSS typed solution for this problem at that time. I cleaned up with a concept of `contract`/`master`/`worker` using a library `simpleWorker` e.g. https://github.com/basarat/tsb/tree/master/src/server/workers/fileListing ... which I plan to back port to atom-ts (or a node module) ... but don't see that happening till TSB is ready for public release. 

FWIW TSB has a concept of active project (and tsb.json to list projects) that doesn't hit the disk as much and keeps it all in sync :rose:

**Comment by ralfbiedert** [comment]
@blakeembrey: Ahhh, yes, mixed it up. Now that I changed it to `exclude` it seems to work here. Thanks!
